samza-api/src/main/java/org/apache/samza/operators/functions/ScheduledFunction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.functions;
+
+import org.apache.samza.operators.Scheduler;
+
+import java.util.Collection;
+
+
+/**
+ * Allows scheduling with key(s) and is invoked when the specified time(s) occurs.
```

Comment:
Maybe: "Allows scheduling an callback for a specific epoch-time."

In general, I'd prefer to emphasize the "scheduling an epoch-time callback" aspect of the API and de-emphasize the "scheduling with/for a key" aspect of the API in docs. IMHI the main feature is the ability to schedule callbacks for future. The per-key per-task scheduling seem like incidental details. Calling out the epoch-time part is important to avoid mistakes - typically callbacks are scheduled after an interval, not at wall-clock times.

Reply:
Great suggestion. Even though "key" is the argument, there is still a callback (i.e. onCallback).